src/native/eventpipe/ds-protocol.c

```diff
@@ -453,6 +453,43 @@ ds_ipc_message_try_parse_uint32_t (
 	return result;
 }
 
+bool
+ds_ipc_message_try_parse_string_utf16_t_string_utf8_t_alloc (
+	uint8_t **buffer,
+	uint32_t *buffer_len,
+	ep_char8_t **string_utf8)
+{
+	EP_ASSERT (buffer != NULL);
+	EP_ASSERT (buffer_len != NULL);
+	EP_ASSERT (string_utf8 != NULL);
+
+	bool result = false;
+
+	uint8_t *byte_array = NULL;
+	uint32_t byte_array_len = 0;
+
+	ep_raise_error_if_nok (ds_ipc_message_try_parse_string_utf16_t_byte_array_alloc (buffer, buffer_len, &byte_array, &byte_array_len));
+
+	*string_utf8 = NULL;
```

Comment:
should move before above line or you might end up freeing random value if above fails.

Reply:
Right, thanks for the catch!